src/oscar/apps/checkout/surcharges.py

```diff
@@ -0,0 +1,63 @@
+from decimal import Decimal as D
+
+from oscar.core import prices
+
+
+class Base(object):
+    """
+    Surcharge interface class
+
+    This is the superclass to the classes in surcharges.py. This allows using all
+    surcharges interchangeably (aka polymorphism).
+
+    The interface is all properties.
+    """
+
+    # The name of the surcharge, shown to the customer during checkout
+    name = "Default surcharge"
+
+    description = ""
+
+    charge_is_percentage = False
+
+    charge_excl_tax = None
+    charge_incl_tax = None
+
+    charge_is_percentage = False
```

Comment:
This should go in the implementation instead of bbeing a flag. EG. implement a PercentageSurcharge and a FlatSurcharge that differ in their calculate method instead of having 2 different surcharges in 1 object.